torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -335,6 +361,82 @@ class TORCH_CUDA_CU_API InputsIdLookup : public NonCopyable {
   std::unordered_map<std::string, EncodingEntry> encoding_lookup_;
 };
 
+//! OptimizedFusionGraph is another layer of caching representing
```

Comment:
I also don't know what optimized means here.

Reply:
I think the amount of work cached will increase for sure. When we make the segments input-dependent, we will have to cache a few of the segmented fusions that apply for different input cases. I'd prefer adding a class and call this another layer of caching just for clarity, and I'm hoping to make this layer manage segmented fusion and welford translation.

Reply:
Apologies again for the confusion here because of separating parts out from welford PR (#843). By `OptimizedFusion` I meant `A copy of original fusion that have been performed input-dependent actions on` and actions for now will include segmentation , welford translation, and scheduling. Maybe `InputDependentFusion` would sound clearer.